File: sales_warehouse_management_py/app/models/purchase_order.py
from sqlalchemy import Column, String, Boolean, DateTime, Enum, Numeric, ForeignKey, Text
from sqlalchemy.dialects.postgresql import UUID
import uuid
from datetime import datetime
from app.core.database import Base
import enum

class PurchaseOrderStatus(str, enum.Enum):
    PENDING = "pending"
    APPROVED = "approved"
    RECEIVED = "received"
    CANCELLED = "cancelled"

class PurchaseOrder(Base):
    __tablename__ = "purchase_orders"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    supplier_id = Column(String(255), nullable=False)  # Simplified - could be FK to suppliers table
    order_date = Column(DateTime, default=datetime.utcnow, nullable=False)
    status = Column(Enum(PurchaseOrderStatus), default=PurchaseOrderStatus.PENDING, nullable=False)
    total_amount = Column(Numeric(10, 2), nullable=False)
    notes = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    def __repr__(self):
        return f"<PurchaseOrder {self.id} for supplier {self.supplier_id}>"

File: backend/src/models/PurchaseOrder.js
const { DataTypes } = require('sequelize');

const PurchaseOrder = {
  initModel: (sequelize) => {
    const PurchaseOrderModel = sequelize.define(
      'PurchaseOrder',
      {
        id: {
          type: DataTypes.UUID,
          defaultValue: DataTypes.UUIDV4,
          primaryKey: true,
        },
        po_number: {
          type: DataTypes.STRING,
          allowNull: false,
          unique: true,
        },
        supplier_name: {
          type: DataTypes.STRING,
          allowNull: false,
        },
        po_date: {
          type: DataTypes.DATE,
          defaultValue: DataTypes.NOW,
        },
        expected_delivery_date: {
          type: DataTypes.DATE,
          allowNull: true,
        },
        status: {
          type: DataTypes.ENUM('pending', 'confirmed', 'received', 'cancelled'),
          defaultValue: 'pending',
        },
        total_amount: {
          type: DataTypes.DECIMAL(15, 2),
          allowNull: false,
        },
        tax_amount: {
          type: DataTypes.DECIMAL(15, 2),
          defaultValue: 0,
        },
        notes: {
          type: DataTypes.TEXT,
          allowNull: true,
        },
        created_at: {
          type: DataTypes.DATE,
          defaultValue: DataTypes.NOW,
        },
        updated_at: {
          type: DataTypes.DATE,
          defaultValue: DataTypes.NOW,
        },
      },
      {
        tableName: 'purchase_orders',
        timestamps: false,
        underscored: true,
      }
    );
    return PurchaseOrderModel;
  },
};

module.exports = PurchaseOrder;
